src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookMultibranchTrigger.java

```diff
@@ -16,26 +12,72 @@
 import org.kohsuke.stapler.DataBoundConstructor;
 
 import javax.inject.Inject;
-import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import static java.util.Objects.requireNonNull;
-
 public class BitbucketWebhookMultibranchTrigger extends Trigger<MultiBranchProject<?, ?>> {
 
+    //the version (of this class) where the PR trigger was introduced. Version is 0 based.
+    private static final int BUILD_ON_PULL_REQUEST_VERSION = 1;
     private static final Logger LOGGER = Logger.getLogger(BitbucketWebhookMultibranchTrigger.class.getName());
 
+    private final boolean pullRequestTrigger;
+    private final boolean refTrigger;
+    /**
+     * This exists as a simple upgrade task. Old classes will de-serialise this to default value (of 0). New
+     * classes will serialise the actual value that was stored. Because the constructor is not run during de-serialisation
+     * we can safely set the value in the constructor to indicate which version this class is.
+     *
+     * @since 3.0.0
+     */
+    private final int version;
+
     @SuppressWarnings("RedundantNoArgConstructor") // Required for Stapler
     @DataBoundConstructor
-    public BitbucketWebhookMultibranchTrigger() {
+    public BitbucketWebhookMultibranchTrigger(boolean pullRequestTrigger, boolean refTrigger) {
```

Comment:
I'm still a bit naff about this, but it's not a PR blocker. We should revisit if we add any more triggers in the future.

Reply:
I agree, but seeing as this is the DataboundConstructor we must have a field for every option in the UI, so.. at least here I think we need to have the listing.